runcoms/zshenv

```diff
@@ -6,6 +6,6 @@
 #
 
 # Ensure that a non-login, non-interactive shell has a defined environment.
-if [[ "$SHLVL" -eq 1 && ! -o LOGIN && -s "${ZDOTDIR:-$HOME}/.zprofile" ]]; then
+if [[ "$SHLVL" -eq 1 && ! -o LOGIN && -s "${ZDOTDIR:-$HOME}/.zprofile" || ! "${TMPDIR}" ]]; then
```

Comment:
There's another issue somewhere about changing this check because it doesn't completely make sense... but what you mentioned should be good for now (I don't know what happened to those other changes)